extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java

```diff
@@ -636,6 +641,32 @@ void createSecurityHttpUpgradeCheck(Capabilities capabilities, BuildProducer<Syn
         }
     }
 
+    @BuildStep
+    void addTracesSupport(Capabilities capabilities,
+            BuildProducer<TelemetrySupportBuilderCustomizerBuildItem> builderProducer) {
+        if (capabilities.isPresent(Capability.OPENTELEMETRY_TRACER)) {
+            builderProducer.produce(new TelemetrySupportBuilderCustomizerBuildItem(new TracesBuilderCustomizer()));
+        }
+    }
+
+    @BuildStep
+    @Record(RUNTIME_INIT)
+    SyntheticBeanBuildItem createTelemetrySupportProvider(
+            List<TelemetrySupportBuilderCustomizerBuildItem> builderCustomizerItems, WebSocketTelemetryRecorder recorder) {
+        var builderCustomizers = builderCustomizerItems.stream().map(i -> i.builderCustomizer).toList();
```

Comment:
I think that we should produce a `SyntheticBeanBuildItem` only if `!builderCustomizers.isEmpty()`. At runtime, you can `@Inject Instance<TelemetrySupportProvider> instance` and only perform some logic if `instance.isResolvable() == true`.

Reply:
I did it this way to avoid null checks everywhere (or to create that NOOP anyway just after checking that is is not in the CDI). Anyway, I'll do as you wish.

Reply:
Ok, I understand but in general we should try to avoid registering unnecessary beans, or useless runtime components respectively.

Reply:
To conserve resources obviously ;-)